### ISSUE TITLE
[Task] Implement cash package profit comparison

### ISSUE THREAD

**Issue by sweatpotato13** [sweatpotato13/erinn.me#229]
### Goal

Implement a page that compares the estimated profit or loss from purchasing the three 2026 Chuseok cash packages and selling their contents on the auction house, using a cash-to-gold conversion rate entered by the user. The flow is **enter a conversion rate → compare three packages → adjust the selected package's item prices**. Communicate the workflow through visual hierarchy, layout, input controls, and selection states rather than lengthy explanations. Preserve erinn.me's existing font, light theme, panels, item images, and accent colors.

The issue is written in English, but the product UI remains Korean. Korean item names and UI strings below are intentional and must not be replaced with unofficial English translations.

#### Confirmed user requirements

- Value color potions using the lowest unit price across all colors multiplied by quantity. Do not calculate probability-weighted values for individual colors.
- Do not display a "gold recovered per 10,000 cash" metric.
- Treat failed lookups and items without listings as 0 gold, allow manual unit-price entry, and make the unresolved state identifiable separately.
- Avoid middle-dot separators wherever possible. Do not repeat descriptions, help text, or badges for every feature.
- Match the site's existing visual language. Use design to communicate interactions while retaining essential labels and accessibility.

#### Page and UI

- Route: `/tools/cash-packages`. Korean page name: `캐시 패키지 비교`.
- Add the page to the existing `생활·파티 계산` group in `FEATURE_LINKS`. Follow existing metadata and sitemap conventions.
- Header: title, refresh action, conversion-rate input labeled `환산 기준` (`10,000 캐시 = [input]만 G`), and membership toggle. Do not prefill an arbitrary conversion rate.
- Desktop: show three comparison cards side by side in the fixed order Sodamhan (`소담한`), Dalgoun (`달고운`), and Onnuri (`온누리`). Selecting a card updates the contents panel below. Select Sodamhan initially; refreshing prices must not reorder the cards.
- Cards: show the product image and name, cash price, prominently sized estimated profit/loss (`예상 손익`), a smaller return percentage, converted cost (`환산 비용`), and estimated net proceeds (`예상 수령`). Show gross sales and fees in the settlement area below the contents instead of repeating them in every card.
- Indicate the selected card using the existing accent border and lightly tinted background. Distinguish profit and loss with +/- signs as well as color. Do not add guaranteed-sale language or purchase recommendation badges.
- Contents: game icon, item name, sale quantity, editable unit price, and subtotal. Allow adjustment of the selected package's purchase quantity. Update the card's cost and available contents consistently with that quantity.
- Allow a sale quantity of 0 to exclude an item. For selection boxes, provide an inline red/blue choice and allow quantity allocation between them when needed. Initially select the option with the higher observed minimum listing price, but do not recommend an option with an unknown price as superior. Refreshing must not overwrite manual selections.
- Edit prices directly in their input fields. Show a small restore button only for manually overridden fields. Refresh preserves manual prices; restore applies the latest automatic price.
- Use skeletons in numeric areas while loading. Display `0` and a warning icon for failed lookups or missing listings. Clicking the icon or opening it with the keyboard reveals the cause and a retry action.
- Show a short badge such as `미확인 2` only when needed. Activating it navigates to that package's unresolved items. Do not add automatic/loaded badges to every normal row.
- Render color potions as ordinary item rows. Expose the pricing basis, `전체 색상 최저가`, only in that row's secondary information.
- Settlement area: gross sales, fees, and estimated net proceeds. Reveal coupon-use settings and calculation details on demand.
- Mobile: show the three products as compact stacked rows. Put the product name and cash price on the left, and profit/loss and return percentage on the right. Place quantity and unit-price controls below each item name to avoid horizontal scrolling.
- Reuse the existing 1280px maximum width, 12px panel radius, thin borders, theme colors, and MabinogiClassic font. Right-align amounts and use tabular numerals. Do not introduce new brand colors or a separate UI library.
- Provide touch targets of at least 44px. Give icon buttons accessible names and support keyboard operation and selection state for cards. Do not hide essential information in hover-only tooltips.

#### Product data and calculations

Recheck the official product descriptions and usage notes during implementation, then register verified static data. Do not build an automatic crawler. Keep product prices and contents separate from auction-market prices.

| Product | Cash | Contents (official Korean names) |
| --- | ---: | --- |
| Sodamhan (`소담한`) | 32,900 | 낭만 농장 개간권 stages 1–6 ×1 each; 낭만 농장 이전권 ×1; 정령 무기 추가 계약권 ×1; 인벤토리 확장권 ×3; 은행 한도 확장권(1천만) ×5; 소지 골드 한도 확장권(100만 골드) ×2 |
| Dalgoun (`달고운`) | 89,900 | 찬란한 세공 도구 ×40; 기억의 보석 ×40; 환생의 비약 상자 ×10; 보호의 6단계 개조석 선택 상자 ×2; 보호의 7단계 개조석 선택 상자 ×2; 스크롤 전용 해제 포션 ×1; 증폭된 스킬 수련 인장(50) ×5; 경매장 수수료 100% 할인 쿠폰 ×1; 인챈트 추출 보호 포션 ×1 |
| Onnuri (`온누리`) | 119,000 | 찬란한 세공 도구 ×80; 기억의 보석 ×80; 반짝이 이름/채팅 지정 색상 변경 포션 상자 ×5; 간편한 망각의 비약 ×20; 환생의 비약 상자 ×20; 인챈트 보호 포션 ×2; 보호의 6단계 개조석 선택 상자 ×2; 보호의 7단계 개조석 선택 상자 ×2; 특별 개조 전환 키트 ×1; 간편한 특별 개조 복구의 정수 ×1; 전용 해제 포션 ×1; 스크롤 전용 해제 포션 ×2; 특별한 유물 조각(1000%) ×10; 증폭된 스킬 수련 인장(50) ×10; 경매장 수수료 100% 할인 쿠폰 ×3; 인챈트 추출 보호 포션 ×2 |

- Packages themselves are untradeable. Ordinary contents can be traded once. Boxes are untradeable, but their resulting items can be traded once, so look up the actual saleable contents.
- One `환생의 비약 상자` yields one `환생의 비약`. Protective upgrade-stone selection boxes yield a red or blue stone of the same stage. Color-potion boxes yield `반짝이 이름/채팅 지정 색상 변경 포션(30일)`.
- Default to the current lowest listing price. Match ordinary items by exact identity and relevant options. Only color potions aggregate color variants of the same product; do not mix different durations or other potion types.
- `converted gold cost = package cash price × purchase quantity ÷ 10,000 × reference gold amount`.
- `gross sales = Σ(sale quantity × effective unit price)`.
- `estimated net proceeds = gross sales − sum of fees for individual sales`.
- `estimated profit/loss = estimated net proceeds − converted gold cost`.
- `return percentage = estimated profit/loss ÷ converted gold cost × 100`.
- The conversion-rate input uses units of 10,000 gold (`만 G`); convert it to actual gold for calculations. If the rate is blank, still show sales values, but show `—` for converted cost, profit/loss, and return percentage.
- Reuse existing fee rules (5% standard, 4% with membership, integer truncation per sale) and verify that they match current game rules.
- Sell the package's 100% auction-fee discount coupons by default. If a coupon is used instead, remove it from the quantity being sold and waive the fee only for its assigned sale. Never waive fees for the entire package with one coupon or count both coupon-sale proceeds and the savings from using that same coupon.
- Verify which items can be sold in stacks and how many actual sales are involved. Coupon-use settings must distinguish the amount/quantity of individual sales. Split sales must consume coupons and calculate fees separately. Do not assume all package contents constitute a single sale.

### Non-goals

- Per-color potion selection, probability-weighted values, or box-opening simulation.
- A gold-recovered-per-10,000-cash metric.
- Automatic exchange-rate collection, real-money trading integrations, or actual purchase/sale actions.
- A recent-transaction-price mode, historical charts, price alerts, or purchase-budget optimization.
- An admin interface, automatic product collection, a database, or new UI dependencies.
- A site-wide redesign or unrelated refactoring.
- Descriptions, tooltips, or badges for every interaction; long introductions or large event banners.

### Code Anchors

Based on the current `develop` branch. Use the proposed new files below as a starting point and prefer reusing existing modules.

- Add: `src/app/tools/cash-packages/page.tsx` — page and metadata.
- Add: `src/app/tools/cash-packages/cash-package-tool.tsx` — comparison cards, contents editing, and responsive UI.
- Add: `src/data/cash-packages.json` — product IDs, prices, contents, box-output mappings, and official sources.
- Add: `src/lib/cash-packages.ts` — pure profit/loss calculations and input validation.
- Modify: `src/lib/feature-links.ts:1` `FEATURE_LINKS` — navigation and homepage entry points.
- Inspect/modify if needed: `src/app/sitemap.ts` — expose the new page.
- Reference: `src/components/tools/preparation.module.css:1` `.page`, `.panel`, `.good`, `.stepper` — existing visual patterns.
- Reference: `src/styles/globals.css` and `src/app/layout.tsx` — font and light theme.
- Reference: `src/app/tools/crafting/crafting-ui.tsx` `NumberField`, `Quantity` — input patterns. Do not copy that screen's lengthy explanatory text.
- Reference: `src/app/calculator/auction-calculator.tsx` — fee UI and manual-price state handling.
- Reference: `src/lib/auction-calculator.ts:147` `calculateAuctionDistribution` — fee rules. Reuse only the shared calculations needed without breaking existing party distribution behavior.
- Reference: `src/lib/api/auction-market.ts:53` `fetchCurrentItemMarket`, `:137` `getCachedCurrentItemMarket` — market lookup and caching. Do not treat cached general name-search results as exact identity matches.
- Reference: `src/app/api/auction/price-summary/route.ts` `GET`, `src/lib/api/auction.ts` — existing proxy and client calls.
- Reference: `src/lib/auction-market.ts:29` `prepareAuctionResults` — valid listings and unit-price handling.
- Reference: `src/lib/utils/index.ts` `getItemImageUrl`, `src/lib/auction-url.ts` — item images and auction links.
- Add test: `src/lib/__tests__/cash-packages.test.ts`.
- Add test: `src/app/tools/cash-packages/__tests__/cash-package-tool.test.tsx`.
- If market API changes are necessary, preserve existing callers' behavior and add regression coverage to the relevant API tests. Keep API keys server-side.

### Interface Contract

The following is the minimum contract for calculation results and price state. Reuse existing types where they express the same contract; do not add unnecessary abstractions.

```typescript
type PriceState = {
  status: 'loading' | 'available' | 'empty' | 'error';
  marketUnitGold: number | null;
  manualUnitGold: number | null; // 0 is a valid override
  fetchedAt: string | null;
  isComplete: boolean;
};

type PackageResult = {
  cashCost: number;
  goldCost: number | null;
  grossGold: number;
  feeGold: number;
  netGold: number;
  profitGold: number | null;
  profitPercent: number | null;
  unpricedCount: number;
};

// Once loading is settled:
// unitGold = manualUnitGold ?? marketUnitGold ?? 0
// netGold = grossGold - feeGold
// profitGold = goldCost === null ? null : netGold - goldCost
// unpricedCount includes only included sale items without a usable price
// and without a manual override. A manual value of 0 resolves the warning.
```

Deduplicate lookups for the same item and share results across packages.

### Acceptance Criteria

- [ ] `pnpm exec jest --runTestsByPath src/lib/__tests__/cash-packages.test.ts src/app/tools/cash-packages/__tests__/cash-package-tool.test.tsx` — all pass.
- [ ] `pnpm test -- --runInBand` — existing test suite passes.
- [ ] `pnpm typecheck` — passes.
- [ ] `pnpm lint` — passes.
- [ ] Pure calculation example: a 32,900-cash package, a rate of 10,000 cash = 10,000,000G, gross sales of 38,000,000G, a standard 5% fee, and no coupons produce a converted cost of 32,900,000G, net proceeds of 36,100,000G, and profit of +3,200,000G.
- [ ] Cover membership, increased purchase quantity, zero sale quantity, allocation between selection-box outputs, coupon-use limits, prevention of selling and using the same coupon, and per-sale fee truncation.
- [ ] Cover a blank conversion rate, lookup failure, no listings, a manual price of 0, preservation of manual values after refresh, and restoration of automatic prices.
- [ ] Color potions use the lowest unit price across colors of the same product and exclude other products/durations.
- [ ] All three packages' contents and quantities match official sources, with no double counting of unopened boxes and their outputs.
- [ ] (manual) At widths of 1440px, 768px, and 390px, the font, colors, and panels match the existing site, with no horizontal scrolling.
- [ ] (manual) Users can select a card, edit a unit price, and see profit/loss update without a separate explanation. All three packages remain easy to compare on mobile.
- [ ] (manual) Cards, inputs, restore actions, warning information, and fee settings are keyboard-accessible. Profit/loss, selection, and warnings are not conveyed by color alone.
- [ ] (manual) Middle-dot separators and repetitive explanations are minimized. Neither the gold-per-10,000-cash metric nor a color-probability UI is present.
- [ ] (manual) Unresolved items contribute 0 gold and can be located through the card's unresolved-item badge.

### Edge Cases

- Do not present an in-progress refresh as an actual 0-gold result. Preserve previous data or show a distinct loading state. Once a lookup fails, apply the requested 0-gold fallback and retain its failure state.
- Validate blank inputs, negative values, NaN, Infinity, fractional quantities, and amounts/products outside the safe integer range. The conversion rate must be positive; never divide by zero.
- Sale quantities plus coupon-use quantities must not exceed available contents. Preserve manual unit prices when purchase quantity changes and consistently enforce quantity limits.
- Do not present a lookup stopped by a pagination cap as a confirmed market-wide minimum. The observed price may be used, but secondary status information must identify partial results.
- Do not combine similarly named variants, durations, or options. Aggregation across color options for the specified potion is the explicit exception.
- Even when sale quantities increase, results remain estimates based on current minimum listing prices. Do not imply guaranteed execution or sale of the entire quantity.
- A single HTTP failure must not clear all other package/item results. Share duplicate item lookups and follow existing server-cache and request-limit patterns.
- Keep calculations in actual gold separate from display formatting in units of 10,000 gold. Do not recalculate profit/loss from rounded display values.

### Dependencies

- Template: `.github/ISSUE_TEMPLATE/ai-task.md` (`fba9f56`).
- Follow repository `AGENTS.md`: before writing code, read the relevant guides in the installed Next.js package's `node_modules/next/dist/docs/`.
- Official products: [Sodamhan](https://mabinogi.nexon.com/ItemShop/product_detail.asp?product_no=630515), [Dalgoun](https://mabinogi.nexon.com/ItemShop/product_detail.asp?product_no=630514), [Onnuri](https://mabinogi.nexon.com/ItemShop/product_detail.asp?product_no=630512).
- Official trading conditions and box details: [2026 Chuseok gift sets](https://mabinogi.nexon.com/page/news/notice_view.asp?id=4893855).
- Existing UI references: `/calculator`, `/tools/crafting`, `/tools/barter`.
- No separately approved high-fidelity mockup exists. Implement against this issue's UI specification and confirmed user requirements, then manually verify the final responsive screens.
